### ISSUE TITLE
numpy.core._exceptions._UFuncNoLoopError: ufunc did not contain a loop with signature matching types

### ISSUE THREAD

**Issue by aedeheathers1002** [librosa/librosa#1732]
***BEFORE POSTING A BUG REPORT*** Please look through [existing issues (both open and closed)](https://github.com/librosa/librosa/issues?q=is%3Aissue) to see if it's already been reported or fixed!

**Describe the bug**
Getting error running librosa.

```
>>> tempo, beat_frames = librosa.beat.beat_track(y=y, sr=sr)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ubuntu/.pyenv/versions/3.10.12/lib/python3.10/site-packages/librosa/beat.py", line 177, in beat_track
    bpm = _tempo(
  File "/home/ubuntu/.pyenv/versions/3.10.12/lib/python3.10/site-packages/librosa/feature/rhythm.py", line 418, in tempo
    tg = tempogram(
  File "/home/ubuntu/.pyenv/versions/3.10.12/lib/python3.10/site-packages/librosa/feature/rhythm.py", line 176, in tempogram
    autocorrelate(odf_frame * ac_window, axis=-2), norm=norm, axis=-2
  File "/home/ubuntu/.pyenv/versions/3.10.12/lib/python3.10/site-packages/librosa/core/audio.py", line 938, in autocorrelate
    powspec = util.abs2(fft.rfft(y, n=n_pad, axis=axis))
  File "/home/ubuntu/.pyenv/versions/3.10.12/lib/python3.10/site-packages/librosa/util/utils.py", line 2514, in abs2
    y = _cabs2(x)
numpy.core._exceptions._UFuncNoLoopError: ufunc '_cabs2' did not contain a loop with signature matching types <class 'numpy.dtype[complex128]'> -> None
```

and

```
>>> y_shifted = librosa.effects.pitch_shift(y=y, sr=sr, n_steps=4)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ubuntu/.pyenv/versions/3.10.12/lib/python3.10/site-packages/librosa/effects.py", line 336, in pitch_shift
    time_stretch(y, rate=rate, **kwargs),
  File "/home/ubuntu/.pyenv/versions/3.10.12/lib/python3.10/site-packages/librosa/effects.py", line 245, in time_stretch
    stft_stretch = core.phase_vocoder(
  File "/home/ubuntu/.pyenv/versions/3.10.12/lib/python3.10/site-packages/librosa/core/spectrum.py", line 1457, in phase_vocoder
    d_stretch[..., t] = util.phasor(phase_acc, mag=mag)
  File "/home/ubuntu/.pyenv/versions/3.10.12/lib/python3.10/site-packages/librosa/util/utils.py", line 2602, in phasor
    z = _phasor_angles(angles)
numpy.core._exceptions._UFuncNoLoopError: ufunc '_phasor_angles' did not contain a loop with signature matching types <class 'numpy.dtype[float32]'> -> None
```

**To Reproduce**
I'm just following this guide. I get error on the line `librosa.beat.beat_track(y=y, sr=sr)`
```
# Beat tracking example
import librosa

# 1. Get the file path to an included audio example
filename = librosa.example('nutcracker')


# 2. Load the audio as a waveform `y`
#    Store the sampling rate as `sr`
y, sr = librosa.load(filename)

# 3. Run the default beat tracker
tempo, beat_frames = librosa.beat.beat_track(y=y, sr=sr)

print('Estimated tempo: {:.2f} beats per minute'.format(tempo))

# 4. Convert the frame indices of beat events into timestamps
beat_times = librosa.frames_to_time(beat_frames, sr=sr)
```

**Expected behavior**
No error.

**Screenshots**
None.

**Software versions***
```
import platform; print(platform.platform())
Linux-5.15.0-1038-aws-x86_64-with-glibc2.31

import sys; print("Python", sys.version)
Python 3.10.12 (main, Jul 24 2023, 00:51:38) [GCC 9.4.0]

import numpy; print("NumPy", numpy.__version__)
NumPy 1.22.0

import scipy; print("SciPy", scipy.__version__)
SciPy 1.11.1

import librosa; print("librosa", librosa.__version__)
librosa 0.10.0

librosa.show_versions()
INSTALLED VERSIONS
------------------
python: 3.10.12 (main, Jul 24 2023, 00:51:38) [GCC 9.4.0]

librosa: 0.10.0

audioread: 3.0.0
numpy: 1.22.0
scipy: 1.11.1
sklearn: 1.3.0
joblib: 1.3.1
decorator: 5.1.1
numba: 0.57.0
soundfile: 0.12.1
pooch: v1.7.0
soxr: 0.3.5
typing_extensions: installed, no version number available
lazy_loader: installed, no version number available
msgpack: 1.0.5

numpydoc: None
sphinx: None
sphinx_rtd_theme: None
matplotlib: 3.7.2
sphinx_multiversion: None
sphinx_gallery: None
mir_eval: None
ipython: None
sphinxcontrib.rsvgconverter: None
pytest: None
pytest_mpl: None
pytest_cov: None
samplerate: None
resampy: None
presets: None
packaging: 23.1
```

**Additional context**

**Comment by aedeheathers1002** [comment]
Solved by doing `pip install --upgrade numpy=1.23.5`. I had numpy 1.22.

**Comment by bmcfee** [comment]
Numpy 1.22 should work, with a few exceptions for broken minor versions that are explicitly coded out in the dependencies for librosa 0.10.0.post1 and later. But 1.23 will entirely avoid the problem as well.
